modules/zivid/presets.py

```diff
@@ -0,0 +1,144 @@
+"""Contains the Preset and Category classes."""
+import _zivid
+
+from zivid.settings import _to_settings
+from zivid.camera_info import CameraInfo
+
+
+class Preset:
+    """3D settings preset.
+
+    Presets are pre-defined settings that are tuned for different camera models to perform optimally
+    under different conditions and use cases.
+
+    This class cannot be initialized directly by the end-user. Use the presets method on the
+    Category class to obtain a Camera instance.
+    """
+
+    def __init__(self, impl):
+        """Initialize Preset wrapper.
+
+        This constructor is only used internally, and should not be called by the end-user.
+
+        Args:
+            impl:   Reference to internal/back-end instance.
+
+        Raises:
+            TypeError: If argument does not match the expected internal class.
+        """
+        if not isinstance(impl, _zivid.presets.Preset):
+            raise TypeError(
+                "Unsupported type for argument impl. Got {}, expected {}".format(
+                    type(impl), type(_zivid.presets.Preset)
+                )
+            )
+        self.__impl = impl
+
+    @property
+    def name(self):
+        """Get the name of the preset.
+
+        Returns:
+            The name of the preset.
+        """
+        return self.__impl.name()
+
+    @property
+    def settings(self):
+        """Get the settings of the preset.
+
+        The settings might change between different releases of the SDK. New presets might be added
+        and old ones might be removed. If having the exact same settings is desired, it is
+        recommended to save them to a YML file and load as needed.
+
+        Returns:
+                The settings of the preset.
+        """
+        return _to_settings(self.__impl.settings())
+
+    def __str__(self):
+        return str(self.__impl)
+
+
+class Category:
+    """Preset category.
+
+    This class cannot be initialized directly by the end-user. Use the categories function to
+    obtain a list of categories available for a camera.
+    """
+
+    def __init__(self, impl):
+        """Initialize Category wrapper.
+
+        This constructor is only used internally, and should not be called by the end-user.
+
+        Args:
+            impl:   Reference to internal/back-end instance.
+
+        Raises:
+            TypeError: If argument does not match the expected internal class.
+        """
+        if not isinstance(impl, _zivid.presets.Category):
+            raise TypeError(
+                "Unsupported type for argument impl. Got {}, expected {}".format(
+                    type(impl), type(_zivid.presets.Category)
```

Comment:
`type(_zivid.presets.Category)` is similarly wrong here.

Reply:
Also, consider using fstrings instead of string.format

Reply:
See the other similar comment for my opinion.

Regarding f-strings: We currently have no f-strings in this package at all. The reason is that we used to support Python 3.5, which doesn't have f-strings. Let's just keep it consistent for now, and then we can upgrade to f-strings later.